docs/tools/firmware.sh

```diff
@@ -577,49 +577,58 @@ case "$os" in
 				echo -e "\nDone!"
 				;;
 			(2)
-				# Detect whether dmg2img are installed
 				echo -e "\nChecking for missing dependencies"
-				dmg2img_check
 				# Load the apfs driver, and install if missing
-				sudo modprobe apfs >/dev/null 2>&1 || apfs_install
+				sudo modprobe ${verbose} apfs 2>/dev/null || apfs_install
 				unmount_macos_and_cleanup () {
 					sudo rm -r ${verbose} ${workdir} || true
-					sudo umount ${verbose} ${loopdevice} || true
-					sudo umount ${verbose} ${macosvol} || true
-					sudo rm -r ${verbose} ${imgdir} || true
+					for i in 0 1 2 3 4 5
```

Comment:
If the user has made extra apfs volumes could this cause issues? I think i have a couple extra volumes so I'll test it when I boot to Linux in a minute.

Reply:
I had tried making extra Volumes, they get the number from 6. Still, it would be better to test it from your side.

Reply:
hmm seems to work fine for me

Reply:
> hmm seems to work fine for me

Any error you encountered if you run it with -v

Reply:
just errors about some of the apfs volumes not being mountable but that's expected

Reply:
Thanks.